repos/system_upgrade/el7toel8/actors/bindupdate/libraries/updates.py

```diff
@@ -0,0 +1,86 @@
+from leapp.libraries.common import isccfg
+from leapp.libraries.stdlib import run
+
+# Callback for walk function
+callbacks = {
+    'dnssec-lookaside': isccfg.ModifyState.callback_comment_out,
+}
+
+
+def paths_from_issues(issues):
+    """ Extract paths from list of BindConfigIssuesModel """
+    paths = []
+    for issue in issues:
+        paths.append(issue.path)
+    return paths
+
+
+def parser_file(parser, path):
+    for cfg in parser.FILES_TO_CHECK:
+        if cfg.path == path:
+            return cfg
+    return None
+
+
+def debug_log(log, text):
+    if log is not None:
+        log.debug(text)
+
+
+def make_backup(path, backup_suffix='.leapp'):
+    """ Make backup of a file before modification """
+    backup_path = path + backup_suffix
+    run(['cp', '--preserve=all', path, backup_path])
+
+
+def update_section(parser, section):
+    """ Modify one section
+    :ptype section: ConfigSection
+    """
+    state = isccfg.ModifyState()
+    parser.walk(section, callbacks, state)
+    state.finish(section)
+    return state.content()
+
+
+def update_config(parser, cfg):
+    """ Modify contents of file accoriding to rules
+    :ptype cfg: ConfigFile
+    :returns str: Modified config contents
+    """
+    return update_section(parser, cfg.root_section())
+
+
+def update_file(parser, path, log=None, write=True):
+    """ Prepare modified content for the file, make backup and rewrite it
+    :param parser: IscConfigParser
+    :param path: String with path to a file
+    :param log: Log instance with debug(str) method or None
+    :param write: True to allow file modification, false to only return modification status
+    """
+    cfg = parser_file(parser, path)
+    modified = update_config(parser, cfg)
+    if modified != cfg.buffer:
+        debug_log(log, '{0} needs modification'.format(path))
```

Comment:
use logger specific str formatting

Reply:
btw the way the most useful article for me explaining why it should be passed to the logger instead of str is here https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html

Reply:
I would like instead link to logger methods supported in LEAPP. I had trouble finding it. Could you point me to logging classes docs?

Reply:
from leapp.libraries.stdlib import api
api.current_logger

Reply:
No need to use logger specific str formatting, because in our codebase we have a lot of similar examples. However, I would use a proper way:)

Reply:
Feel free to close